test/Base_Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -319,6 +319,11 @@ add_specs suite_builder = suite_builder.group "Dataflow Warnings" group_builder-
         Warning.get_all res wrap_errors=True . map .value . should_equal [(Map_Error.Error 3 (Map_Error.Error 3 3)), (Map_Error.Error 3 (Map_Error.Error 2 2)), (Map_Error.Error 3 (Map_Error.Error 1 1)), (Map_Error.Error 3 (Map_Error.Error 0 0)), (Map_Error.Error 2 (Map_Error.Error 2 2)), (Map_Error.Error 2 (Map_Error.Error 1 1)), (Map_Error.Error 2 (Map_Error.Error 0 0)), (Map_Error.Error 1 (Map_Error.Error 1 1)), (Map_Error.Error 1 (Map_Error.Error 0 0)), (Map_Error.Error 0 (Map_Error.Error 0 0))]
         Warning.get_all (res.at 2) . map .value . should_equal [3, 2, 1, 0, 2, 1, 0, 1, 0, 0]
 
+    group_builder.specify "should wrap errors with even deeper nesting" <|
+        Warning.get_all [[[Warning.attach "warning" 10]]] wrap_errors=True . map .value . should_equal [Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 "warning"))]
+        Warning.get_all [[[[Warning.attach "warning" 10]]]] wrap_errors=True . map .value . should_equal [Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 "warning")))]
+        Warning.get_all [[[[[Warning.attach "warning" 10]]]]] wrap_errors=True . map .value . should_equal [Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 (Map_Error.Error 0 "warning"))))]
```

Comment:
curious: why was this test added?

Did it reproduce the original issue without the convoluted way I originally found? Or is it something else?

Reply:
I added this just for extra coverage; I had no reason to think it would fail. But (as I discovered when implementing this in the first place) the warning propagation is very dependent on complex details of specialization, so I want to cover as many cases as possible, including ones that seem like they wouldn't be an issue.